main.js

```diff
@@ -620,6 +654,120 @@ app.post('/delete-admin', async function(req, res) {
 	}
 });
 
+app.get('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	const filterQuery = {'_id': new mdb.ObjectId(req.session._id)};
+	const user = await db.collection('BBY-6_users').findOne(filterQuery);
+	if(!user) {
+		// Could not find user
+		res.status(500).send("userNotFound");
+		return;
+	}
+	res.status(200).send(user.kits);
+})
+
+app.post('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	try {
+		const filterQuery = {'name': req.body.name};
+		const kitTemplate = await db.collection('BBY-6_kit-templates').findOne(filterQuery);
+		if(!kitTemplate) {
+			res.status(500).send("kitTemplateNotFound");
+			return;
+		}
+		try {
+			const userFilterQuery = {'_id': new mdb.ObjectId(req.session._id)};
+			kitTemplate.kit = kitTemplate.kit.map(element => {
+				element.required = true;
+				element.completed = false;	
+				return element;
+			})
+			const user = await db.collection('BBY-6_users').updateOne(userFilterQuery, {$push: {kits: kitTemplate}});
+			if(!user) {
+				// Could not find user
+				res.status(500).send("userNotFound");
+				return;
+			}
+			res.status(200).send(kitTemplate);
+		} catch(e) {
+			res.status(500).send("serverIssue");
+		}
```

Comment:
I think you don't technically need this nested try/catch block, since the outer `catch` is going to do the same `res.status(500).send("serverIssue")` anyway.